code/game/objects/effects/decals/turfdecal/tilecoloring.dm

```diff
@@ -4,22 +4,22 @@
 	layer = TURF_PLATING_DECAL_LAYER
 	alpha = 110
 
-#define PRIDE_ALPHA 60
+#define DECAL_ALPHA 60
 
+/// Certain holidays have flags/holiday colors set for decals
 /obj/effect/turf_decal/tile/Initialize(mapload)
-	if (check_holidays(APRIL_FOOLS))
+	if(check_holidays(APRIL_FOOLS))
 		color = "#[random_short_color()]"
-	else if (check_holidays(PRIDE_WEEK))
-		var/datum/holiday/pride_week/pride_week = GLOB.holidays[PRIDE_WEEK]
-		color = pride_week.get_floor_tile_color(src)
 
+	else if(check_holidays(PRIDE_WEEK))
+		if(istype(src, /obj/effect/turf_decal/tile/neutral/tram))
+			color = get_holiday_color(src, STRIPE_VERTICAL)
+		else color = get_holiday_color(src, STRIPE_HORIZONTAL)
```

Comment:
`istype(src)` usually is icky behavior, this should probably be a var on the effect that determines whether to use vertical or horizontal stripes

Reply:
Yep, thanks for your help.